src/Compilers/VisualBasic/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.vb

```diff
@@ -5208,5 +5208,43 @@ End Class
             CompileAndVerify(CreateCompilationWithMscorlib45AndVBRuntime(source), symbolValidator:=attributeValidator)
         End Sub
 
+        <Fact, WorkItem(10639, "https://github.com/dotnet/roslyn/issues/10639")>
+        Public Sub UsingStaticDirectiveDoesNotIgnoreObsoleteAttribute()
+            Dim source = <compilation>
+                             <file name="a.vb">
+                                 <![CDATA[
+Imports System
+Imports TestAssembly.TestError
+Imports TestAssembly.TestWarning
+
+<Obsolete("Broken Error Class", True)>
+Public Class TestError
+    Public Shared Sub TestErrorFunc()
+    End Sub
+End Class
+
+<Obsolete("Broken Warning Class", False)>
+Public Class TestWarning
+    Public Shared Sub TestWarningFunc()
+    End Sub
+End Class
+
+Public Module Test
+    Public Sub Main()
+        TestErrorFunc()
+        TestWarningFunc()
+    End Sub
+End Module
+]]>
+                             </file>
+                         </compilation>
+
+            Dim options = New VisualBasicCompilationOptions(OutputKind.ConsoleApplication, rootNamespace:="TestAssembly")
+            Dim compilation = CreateCompilationWithMscorlibAndVBRuntime(source, options)
+
+            compilation.VerifyDiagnostics(
```

Comment:
>compilation.VerifyDiagnostics( [](start = 12, length = 30)

Please always use AssertTheseDiagnostics helper for VB tests, unless we verify the fact that diagnostics is empty.